keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,1084 @@
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Yes, please see testing sections above.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+This change does not affect the API servers. This change may affect the ability
+of a kubelet to run a pod with gitRepo volumes.
+
+If there are no workloads using gitRepo volumes we don't envision any
+disruptions.
+
+If there are workloads using gitRepo volumes then kubelets that have been
+upgraded to have GitRepoVolumeDriver set to false will fail to run pods for
+these workloads.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+Pods failing to start.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+Yes, we are disabling the in-tree gitRepo volume driver. We aren't removing it
+from the API.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+Operators can check if there are pods with gitRepo volumes in their cluster by
+running the following command:
+
+```bash
+kubectl get pods -A -o json \
+| jq -r '.items[] | select(.spec.volumes[]? | has("hostPath")) | .metadata.namespace + " " + .metadata.name'
```

Comment:
this says "hostPath" not "gitRepo"

Tell me you cut-and-pasted without telling me you cut-and-pasted :)

Reply:
haha I didn't have any gitRepo volume containers on the cluster that I was testing that command against so I used hostPath to confirm it worked. good catch!